llvm/lib/Target/RISCV/RISCV.td

```diff
@@ -472,14 +472,22 @@ def HasExtXcvmem
     : Predicate<"Subtarget->hasExtXcvmem()">,
                AssemblerPredicate<(any_of FeatureExtXcvmem),
                "'Xcvmem' (Post-incrementing Load & Store)">;
+// CORE-V Event Load Word Extension
+def FeatureExtXcvelw
+   : SubtargetFeature<"xcvelw", "HasExtXcvelw", "true",
+                      "'Xcvelw' (Event Load Word)">;
+def HasExtXcvelw
+   : Predicate<"Subtarget->hasExtXcvelw()">,
+              AssemblerPredicate<(any_of FeatureExtXcvelw),
+              "'Xcvelw' (Event Load Word)">;
 // All CORE-V extensions
```

Comment:
What about adding this so that it automatically matches against expressions like `a+2`:
```
let Predicates = [HasExtXcvelw] in {
  def : Pat<(int_riscv_cv_elw_elw (AddrRegImm GPR:$rs1, simm12:$imm12)),
            (CV_ELW GPR:$rs1, simm12:$imm12)>;
}
```
Then you can make complicated testcases like this pass:
```
define i32 @test.cv.elw.elw2(i8* %a, i32 %b) {
; CHECK-LABEL: test.cv.elw.elw2:
; CHECK:       # %bb.0:
; CHECK-NEXT:    add a0, a1, a0
; CHECK-NEXT:    cv.elw a0, 7(a0)
; CHECK-NEXT:    ret
  %c = add i32 %b, 4
  %d = add i32 %c, 3
  %e = getelementptr i8, i8* %a, i32 %d
  %1 = call i32 @llvm.riscv.cv.elw.elw(i8* %e)
  ret i32 %1
}
```

Reply:
Addressed you concern.